api/src/test/java/org/pmiops/workbench/api/WorkspacesControllerTest.java

```diff
@@ -2512,6 +2508,63 @@ public void testShareWorkspaceRemoveBillingProjectUser() {
     verify(fireCloudService, times(1)).addOwnerToBillingProject(any(), any());
   }
 
+  @Test
+  public void testShareWorkspacePublishedWorkspace() {
+    DbCdrVersion ctCdrVersion =
+        TestMockFactory.createControlledTierCdrVersion(cdrVersionDao, accessTierDao, 5L);
+
+    DbUser writerUser = createAndSaveUser("writerfriend@gmail.com", 124L);
+
+    when(fireCloudService.getWorkspaceAclAsService(anyString(), anyString()))
+        .thenReturn(
+            createWorkspaceACL(
+                new JSONObject()
+                    .put(
+                        // Specifically, the REGISTERED tier is used for publishing.
+                        accessTier.getAuthDomainGroupEmail(),
+                        new JSONObject()
+                            .put("accessLevel", "READER")
+                            .put("canCompute", false)
+                            .put("canShare", false))
+                    .put(
+                        currentUser.getUsername(),
+                        new JSONObject()
+                            .put("accessLevel", "OWNER")
+                            .put("canCompute", true)
+                            .put("canShare", true))
+                    .put(
+                        writerUser.getUsername(),
+                        new JSONObject()
+                            .put("accessLevel", "WRITER")
+                            .put("canCompute", true)
+                            .put("canShare", true))));
+
+    Workspace workspace =
+        createWorkspace().cdrVersionId(Long.toString(ctCdrVersion.getCdrVersionId()));
+    workspace = workspacesController.createWorkspace(workspace).getBody();
+
+    ShareWorkspaceRequest shareWorkspaceRequest =
+        new ShareWorkspaceRequest()
+            .workspaceEtag(workspace.getEtag())
+            // Removing writer.
+            .addItemsItem(
+                new UserRole().email(LOGGED_IN_USER_EMAIL).role(WorkspaceAccessLevel.OWNER));
+
+    stubFcUpdateWorkspaceACL();
+    workspacesController.shareWorkspace(
+        workspace.getNamespace(), workspace.getName(), shareWorkspaceRequest);
+    verify(fireCloudService)
```

Comment:
a comment like "does not remove RT auth domain" would be helpful context here

Reply:
Added